src/client/dfs/dfs.c

```diff
@@ -3575,6 +3586,16 @@ dfs_update_parent(dfs_obj_t *obj, dfs_obj_t *src_obj, const char *name)
 	return 0;
 }
 
+/* Update a object to a new parent, taking the parent directly */
+void
+dfs_update_parentfd(dfs_obj_t *obj, dfs_obj_t *src_obj, const char *name)
+{
+	oid_cp(&obj->parent_oid, src_obj->oid);
+
+	strncpy(obj->name, name, DFS_MAX_NAME);
```

Comment:
check if name is NULL

Reply:
For a public function I'd agree this was necessary.

Reply:
I think it would still be good to return EINVAL for the sake of consistency, and in case this function ever needs to be public. But I also agree that it's okay since it's private.